### DISCUSSION TITLE
Sigma Rule Alerts (old playbook) are  are grayed out for "Tune Detection"

### DISCUSSION THREAD

**Discussion by sleepingbel** [Security-Onion-Solutions/securityonion#13127] [2.4] [answered]
### Version

2.4.70

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

16 v-CPU

### RAM

48 GB

### Storage for /

250 GB

### Storage for /nsm

1.66 TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hello all,

After the upgrade i wanted to tune a sigma rule that is creating a lot of noise, but the "Tune Detection" is grayed out.
![Knipsel](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/c30113dd-fd4a-4d59-8a25-8c55c07f4bb3)

It is also only for sigma rules, the suricata rules can be tuned
![Knipsel1](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/1e2af2f3-6431-4db9-8aa7-df619fab5911)

The status of all my rules is "OK"
![Knipsel3](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/5a841bfa-aa54-498f-8399-0b9bf8013438)

I looked for a solution in the [documentation](https://docs.securityonion.net/en/2.4/detections.html) but couldn't find it

Regards
Bart

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by defensivedepth** [comment]
This is expected. The old Playbook rules should no longer be active and should not be generating any more alerts.

**Reply by sleepingbel** [reply]
Yes ok but I think you mean this is NOT expected, because the old sigma rules are still generating alerts. How can i solve this.

Even if I want to search the rule in the detection tab with a so_detection.title column query, i can not find the rule.

Could it be the reason that this is a medium severity rule that was active before the upgrade and now only high, critical severity rules are active? The problem is not solved with  activating core++ ruleset in sigmaRulePackages I  tested this.

**Reply by defensivedepth** [reply]
The old Playbook rules should have been backed up and removed during the SOUP upgrade to 2.4.70.

Check `/opt/so/rules/elastalert/playbook/` - are there still rules in there? They should have been backed up to `/nsm/backup/detections-migration/sigma/rules/`

**Reply by sleepingbel** [reply, marked as answer]
all the rules are still in the directory /opt/so/rules/elastalert/playbook/

![Knipsel](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/5b273630-1fa5-41bb-a262-67618a81e7a1)

Nothing is backed up 

![Knipsel1](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/bf32a886-818d-48a9-ae2b-1ace1566a13f)

 I have now manually backed up to /nsm/backup/detections-migration/sigma/rules/ and deleted all rules from /opt/so/rules/elastalert/playbook/ . I will check if the alerts come back

**Reply by defensivedepth** [reply]
Can you review the soup logs? It should have backed those files up and removed them:

`sudo grep -nr Playbook /root/soup.log`

**Reply by sleepingbel** [reply]
Regarding the soup log there was no error registered

![Knipsel3](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/f1b0edbe-66d0-4f7a-a36c-73c6c2449c8b)

But I think that my  manually backed up worked, the alerts are not comming back

```
sudo cp /opt/so/rules/elastalert/playbook/* /nsm/backup/detections-migration/sigma/rules/
sudo rm /opt/so/rules/elastalert/playbook/*
```

**Reply by sleepingbel** [reply]
So I need to tune my rules again.
but is it possible to add a column "author" in the soc -  detections? because now you can not filter on author anymore this was possible in playbook
I was using only the rules of Florian Roth

**Reply by defensivedepth** [reply]
Yes, all of these fields are able to be set as GroupBy - 

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/954732/a2fc77a7-c2d0-418b-9e80-a63dcdd83ed4)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/954732/52e63609-83ad-4fb4-bd71-52676ad2ec06)

**Reply by sleepingbel** [reply]
thx for the help

**Comment by defensivedepth** [comment]
@sleepingbel Did you enable any Playbook Plays after soup? Im just trying to understand why it didnt back them up.

**Reply by sleepingbel** [reply]
Hello Josh,

I did not activate playbook plays after the update, so i do not know what went wrong.

But I had two instances whit the same issue (homelab & workplace testenvironment), both were a standalone installation,

In playbook I had a lot of custom local rules and also a lot of community rules with extra filters on it.

**Reply by defensivedepth** [reply]
Would you mind running the following on one of those instances and posting the output here? I am trying to understand why the cleanup process didnt run on your soup:

`sudo grep -A 1 'playbook:'  /opt/so/saltstack/local/pillar/minions/*`

**Reply by sleepingbel** [reply]
Hello Josh this is the output
```

/opt/so/saltstack/local/pillar/minions/seconion_standalone.sls:playbook:
/opt/so/saltstack/local/pillar/minions/seconion_standalone.sls-    enabled: true

```

**Reply by defensivedepth** [reply]
One last question:  Can you confirm that the following file exists?

`/nsm/backup/detections-migration/sigma/playbook-dump.sql`

What version did you upgrade from?

**Reply by sleepingbel** [reply]
hello Josh,

Only the rules directory is in the sigma directory, so playbook-dump.sql does not exists
![Knipsel](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/2875288b-0cef-4b51-928e-5c69446ad9ff)
Both instances where 2.4.60

homelab is a VM on virtualbox, workplace testenvironment is hardware both system have more recourses than the min specs in the documentation.

**Reply by defensivedepth** [reply]
And soup was successful, no errors? Can you upload the entire soup log from one of the systems? Just a text file.

**Reply by sleepingbel** [reply]
Yes soup was succesfull, here is the log

[soup.log](https://github.com/user-attachments/files/15739083/soup.log)

**Reply by sleepingbel** [reply]
Hey josh,

Did you find someting?

**Reply by defensivedepth** [reply]
No nothing yet.

**Reply by defensivedepth** [reply]
@sleepingbel Do you recall what version you did the initial install from?

**Comment by sleepingbel** [comment]
﻿Yes the original install was 2.4.50Met vriendelijke groeten Bart Van Hees Met vriendelijke groeten Bart Van Hees Op 11 jun. 2024 om 12:25 heeft Josh Brower ***@***.***> het volgende geschreven:﻿
@sleepingbel Do you recall what version you did the initial install from?

**Reply by defensivedepth** [reply]
How did you run soup? Interactively?

**Comment by sleepingbel** [comment]
Hey Josh,I aim not sure what you mean with interactively but I did a ssh to the server and entered the command sudo soup and waited until soup finished Met vriendelijke groeten Bart Van Hees Op 11 jun. 2024 om 13:56 heeft Josh Brower ***@***.***> het volgende geschreven:﻿
How did you run soup? Interactively?

**Reply by defensivedepth** [reply]
Ok thats fine. Have you ever disabled the Playbook module? or modified this setting at all?
`/opt/so/saltstack/local/pillar/minions/$HOSTNAME_standalone.sls`

```
playbook:
  enabled: True
```

I can't seem to reproduce the issue.

**Reply by defensivedepth** [reply]
Did you see this when you ran soup?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/954732/189c84ce-7d0a-460b-bba0-7065bf58561d)

**Reply by defensivedepth** [reply]
And this?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/954732/30e9211d-b6f0-4847-a85e-e4aea60607c0)

**Reply by sleepingbel** [reply]
> Did you see this when you ran soup?
> 
> ![image](https://private-user-images.githubusercontent.com/954732/338682741-189c84ce-7d0a-460b-bba0-7065bf58561d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNjk5MDIsIm5iZiI6MTcxODE2OTYwMiwicGF0aCI6Ii85NTQ3MzIvMzM4NjgyNzQxLTE4OWM4NGNlLTdkMGEtNDYwYi1iYmEwLTcwNjViZjU4NTYxZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMlQwNTIwMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMGJmMGVjMDYzYzkwZTAzMTk1NDJmZjE2MzU3Y2UyZTU4NzU1OWJjZjc0NThmYjM5YThkNzY4NzFhYjljMDVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LwyE87znSfOtH4LcBNly4uNsA65u2t7izDZSKw71Jas)

I did not see this Josh, and I never disabled the playbook module

**Reply by sleepingbel** [reply]
> And this?
> 
> ![image](https://private-user-images.githubusercontent.com/954732/338682914-30e9211d-b6f0-4847-a85e-e4aea60607c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNjk5MzcsIm5iZiI6MTcxODE2OTYzNywicGF0aCI6Ii85NTQ3MzIvMzM4NjgyOTE0LTMwZTkyMTFkLWI2ZjAtNDg0Ny1hODVlLWU0YWVhNjA2MDdjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMlQwNTIwMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNWZhYTFhMjQwNDA4YTQ5YTg4NWYxOTMxYTE2NDgyNTIxN2Q0N2I0ZTQ4YzA3NTI5ZTBjMDliZmU5MjZiNmRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.IfMmoSqfPQ4oG5ZSYYTkfsoNYwWHjBIbqg-CK0jEjF4)

This I have seen

**Reply by defensivedepth** [reply]
Thanks @sleepingbel !
